ee/clickhouse/sql/trends/lifecycle.py

```diff
@@ -1,165 +1,109 @@
 _LIFECYCLE_EVENTS_QUERY = """
-WITH person_activity_including_previous_period AS (
-    SELECT DISTINCT 
-        person_id, 
-        {trunc_func}(events.timestamp) start_of_period 
-
-    FROM events
-        JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi 
-            ON events.distinct_id = pdi.distinct_id
-
-    WHERE team_id = %(team_id)s AND {event_query} {filters}
-
-    GROUP BY 
-        person_id, 
-        start_of_period
-        
-    HAVING 
-        start_of_period <= toDateTime(%(date_to)s) 
-        AND start_of_period >= toDateTime(%(prev_date_from)s)
-
-), person_activity_as_array AS (
-    SELECT DISTINCT 
-        person_id, 
-        groupArray({trunc_func}(events.timestamp)) start_of_period 
-
-    FROM events
-        JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi 
-            ON events.distinct_id = pdi.distinct_id
-
-    WHERE 
-        team_id = %(team_id)s 
-        AND {event_query} {filters}
-        AND toDateTime(events.timestamp) <= toDateTime(%(date_to)s) 
-        AND {trunc_func}(events.timestamp) >= toDateTime(%(date_from)s)
-        
-    GROUP BY person_id
-), periods AS (
-    SELECT 
-        {trunc_func}(toDateTime(%(date_to)s) - number * %(seconds_in_interval)s) AS start_of_period 
-        
-    FROM numbers(%(num_intervals)s)
-)
+    WITH 
+        %(team_id)s AS current_team,
+        %(interval)s AS selected_period,
+        INTERVAL {interval} AS interval_type,
+        toDateTime(%(date_from)s) AS selected_date_from,
+        toDateTime(%(date_to)s) AS selected_date_to,
+        dateTrunc(selected_period, selected_date_from) AS selected_interval_from,
+        dateTrunc(selected_period, selected_date_to) AS selected_interval_to,
+        selected_interval_from - INTERVAL {interval} AS previous_date_from,
+        dateDiff(
+           selected_period, 
+            -- Include the period before the first in the activity consideration
+            previous_date_from, 
+            selected_interval_to
+        ) AS number_of_intervals,
+
+        filtered_events AS (
```

Comment:
Note: Will this cause issues due with distributed tables? We've seen previously that it might.

Reply:
I've been testing on production ™️  and this query does work. However, yes in certain configurations of CTEs and on certain nodes we do end up not seeing some CTEs. I've added an [issue for this](https://github.com/ClickHouse/ClickHouse/issues/29748) but haven't reproduced locally, when I have time I'll try a little harder. For the time when I introduce the `EventQuery` I'll just string substitute where it's needed.

btw are you talking about the distributed table engine? I didn't think we were using that?

Reply:
For reference, looks like [this issue](https://github.com/ClickHouse/ClickHouse/issues/23865) might be relevant regarding CTEs, I'll update my issue